cli/kaos_cli/facades/tests/test_backend_facade.py

```diff
@@ -0,0 +1,368 @@
+        # Arrange
+        self.arrange_test_fixtures(environment, backend, url, token, infra, kubeconfig, user)
+        index = 0
+        cloud_env = self.derive_context(environment)
+
+        # Act
+        context_info = self.facade.get_context_info(cloud_env, index)
+        provider = context_info['provider']
+
+        # Assert
+        self.assertEqual(provider, environment)
+
+    @parameterized.expand(backend_facade_test_inputs)
+    def test_method_get_context_by_index(self, environment, backend, url, token, infra, kubeconfig, user):
+        # Arrange
+        self.arrange_test_fixtures(environment, backend, url, token, infra, kubeconfig, user)
+        index = 0
+
+        cloud_env = self.derive_context(environment)
+
+        context_info = self.facade.get_context_info(cloud_env, index)
+
+        # Act
+        context = self.facade.get_context_by_index([context_info], index)
+
+        # Assert
+        self.assertEqual(context, cloud_env)
+
+    @parameterized.expand(backend_facade_test_inputs)
+    def test_method_jsonify_context_list(self, environment, backend, url, token, infra, kubeconfig, user):
+        # Arrange
+        self.arrange_test_fixtures(environment, backend, url, token, infra, kubeconfig, user)
+
+        cloud_env = self.derive_context(environment)
+
+        # Act
+        context_info = self.facade.jsonify_context_list(cloud_env)
+        provider = context_info[0]['provider']
+
+        # Assert
+        self.assertEqual(provider, environment)
+
+    @parameterized.expand(backend_facade_test_inputs)
+    def test_method_set_context_by_context(self, environment, backend, url, token, infra, kubeconfig, user):
+        # Arrange
+        self.arrange_test_fixtures(environment, backend, url, token, infra, kubeconfig, user)
+
+        self.facade.state_service.create()
+        with open(CONFIG_PATH, 'wb') as f:
+            f.close()
+
+        # Act
+        is_set = self.facade.set_context_by_context(environment)
+
+        # Assert
+        self.assertTrue(is_set)
+
+        # Act
+        is_set = self.facade.set_context_by_context('invalid_environment')
+
+        # Assert
+        self.assertFalse(is_set)
+
+    @parameterized.expand(backend_facade_test_inputs)
+    def test_method_set_context_by_index(self, environment, backend, url, token, infra, kubeconfig, user):
+        # Arrange
+        self.arrange_test_fixtures(environment, backend, url, token, infra, kubeconfig, user)
+
+        valid_index = 0
+        invalid_index = 1
+
+        # Act
+        is_set, current_context = self.facade.set_context_by_index(valid_index)
+
+        # Assert
+        self.assertTrue(is_set)
+        self.assertEqual(current_context, environment)
+
+        # Act
+        is_set, current_context = self.facade.set_context_by_index(invalid_index)
+
+        # Assert
+        self.assertFalse(is_set)
+        self.assertIsNone(current_context)
+
+    def test_remove_build_files(self):
+        pass
```

Comment:
Please mark this test as skipped and document its intent. Example from `pytest`: https://docs.pytest.org/en/latest/skipping.html#skipping-test-functions

Reply:
This unit test was passed before as I intended to complete it at a later period which was forgotten. Now it is completed and need not be skipped.